analyzers/src/SonarAnalyzer.Common/Helpers/ProjectConfigReader.cs

```diff
@@ -0,0 +1,80 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using System.Xml.Serialization;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace SonarAnalyzer.Helpers
+{
+    /// <summary>
+    /// This class reads and encapsulates <see cref="ProjectConfig"/>, exposing only the configuration our analyzers need.
+    /// </summary>
+    internal class ProjectConfigReader
+    {
+        private const string SonarProjectConfigFileName = "SonarProjectConfig.xml";
+
+        private readonly Lazy<ProjectConfig> projectConfig;
+
+        public string AnalysisConfigPath => projectConfig.Value.AnalysisConfigPath;
+        public string FilesToAnalyzePath => projectConfig.Value.FilesToAnalyzePath;
+        public string OutPath => projectConfig.Value.OutPath;
+        public string ProjectPath => projectConfig.Value.ProjectPath;
+        public string TargetFramework => projectConfig.Value.TargetFramework;
+        public ProjectType ProjectType =>
+            Enum.TryParse<ProjectType>(projectConfig.Value.ProjectType, out var result)
+            ? result
+            : ProjectType.Product;
```

Comment:
We know that every single rule will  read this variable and therefore parse the enum. Let's add  `private ProjectType? projectType;` field and parse it only once here with little bit more logic.